File: front/src/types/Pedido.ts
export interface PedidoItem {
  productoId: number;
  nombre: string;
  precio: number;
  cantidad: number;
  subtotal: number;
}

export interface PedidoUsuario {
  id: number;
  nombre: string;
  email: string;
}

export interface Pedido {
  id: number;
  usuario: PedidoUsuario;
  items: PedidoItem[];
  total: number;
  estado: string;
  fecha: string;
}

File: back/src/main/java/com/deporshop/model/Pedido.java
package com.deporshop.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "pedidos")
@Data
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Pedido {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "usuario_id", nullable = false)
    public Usuario usuario;

    @ElementCollection
    @CollectionTable(name = "pedido_items", joinColumns = @JoinColumn(name = "pedido_id"))
    public List<PedidoItem> items = new ArrayList<>();

    @Column(nullable = false)
    public double total;

    @Column(nullable = false)
    public String estado;

    @Column(nullable = false)
    public LocalDateTime fecha;

    public Pedido() {
    }

    @PrePersist
    protected void onCreate() {
        this.fecha = LocalDateTime.now();
        if (this.estado == null) {
            this.estado = "CONFIRMADO";
        }
    }
}
